### PR TITLE
dts: nordic: nrf54h20: Disable USBHS core cache

### PR DESCRIPTION
**Pull request by tmon-nordic**
Configure USBHS core registers as non-cachable to prevent D-Cache from inhibiting volatile accesses to the USBHS core registers.

### CODE REVIEW

File: dts/common/nordic/nrf54h20.dtsi [hunk 1]
```diff
@@ -308,6 +309,13 @@
 				status = "disabled";
 			};
 
+			usbhs_core: memory@2f700000 {
+				compatible = "zephyr,memory-region";
+				reg = <0x2f700000 0x40000>;
+				zephyr,memory-region = "USBHS_CORE";
+				zephyr,memory-attr = <( DT_MEM_ARM(ATTR_MPU_RAM_NOCACHE) )>;
```
[the remark this answers is not in the document]
Comment (hubertmis) [reply dts/common/nordic/nrf54h20.dtsi:316]: Both solutions look good to me. I don't know Zephyr's MPU design well enough to assess which is better.

Please note that similar solution is required for I3C + CAN combined memory range so we should solve both issues in the same way.

**Review by hubertmis — DISMISSED** [review]

File: dts/common/nordic/nrf54h20.dtsi (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmon-nordic) [reply dts/common/nordic/nrf54h20.dtsi]: Changed, but I am not sure where to put `select CPU_HAS_CUSTOM_FIXED_SOC_MPU_REGIONS`. I put it under cpuapp which is likely the only core where it does make sense to access usbhs.

**Review by gmarull — DISMISSED** [review]
lgtm, can be easily extended for CAN/I3C

File: dts/common/nordic/nrf54h20.dtsi (hunk 1 above)
[the remark this answers is not in the document]
Comment (hubertmis) [reply dts/common/nordic/nrf54h20.dtsi]: All other things around USB are valid for App and Radio. So I don't think this entry should be available only for cpuapp. It should be for both cpuapp and cpurad, depending on which has `usbhs status="okay"` in its device tree.

[the remark this answers is not in the document]
Comment (tmon-nordic) [reply dts/common/nordic/nrf54h20.dtsi]: To make this available to cpurad, I believe just `select CPU_HAS_CUSTOM_FIXED_SOC_MPU_REGIONS` would have to be added under `config SOC_NRF54H20_CPURAD`. Not sure if I should worry about empty `mpu_regions`.

**Review by hubertmis — APPROVED** [review]
Looks good

**Review by 57300 — APPROVED** [review]
LGTM, thanks

**Review by gmarull — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**